nautobot/core/models/tree_queries.py

```diff
@@ -11,6 +12,27 @@ class TreeQuerySet(TreeQuerySet_, querysets.RestrictedQuerySet):
     """
     Combine django-tree-queries' TreeQuerySet with our RestrictedQuerySet for permissions enforcement.
     """
+    def ancestors(self, of, *, include_self=False):
```

Comment:
I wonder how difficult it would be to add a couple of test cases for the two main branches of this function? I.e. make sure that `location.objects.with_tree_fields.last().ancestors() == location.objects.without_tree_fields.last().ancestors()` at least?

Reply:
Yes! Note that the former returns a queryset and the latter returns a list. In practice I have not found that this is causing any problems, but I thought it would be good to highlight for you to think about.

Reply:
Theoretically, using another query, we could turn it back into a queryset. I'm not sure this is worth it though.

Reply:
Thanks for pointing that out. I am a bit concerned that we might have existing code here or in apps that's assuming that `.ancestors()` returns a queryset? So it might be worthwhile to construct a queryset here if it doesn't cause too much performance impact.

Reply:
Though as I'm thinking about it, that may be tricky to do in a way that still gives the right ordering...

Reply:
I have done this, it is a little complicated to look at but it works.